sdk/scheduler/src/main/java/com/mesosphere/sdk/debug/PlansTracker.java

```diff
@@ -0,0 +1,297 @@
+package com.mesosphere.sdk.debug;
+
+import com.mesosphere.sdk.http.ResponseUtils;
+import com.mesosphere.sdk.offer.Constants;
+import com.mesosphere.sdk.scheduler.plan.Phase;
+import com.mesosphere.sdk.scheduler.plan.Plan;
+import com.mesosphere.sdk.scheduler.plan.PlanCoordinator;
+import com.mesosphere.sdk.scheduler.plan.PlanManager;
+import com.mesosphere.sdk.scheduler.plan.Step;
+import com.mesosphere.sdk.state.StateStore;
+import com.mesosphere.sdk.state.StateStoreUtils;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+// SUPPRESS CHECKSTYLE MultipleStringLiteralsCheck
+
+/**
+ * PlansTracker is the backend of the PlansDebugEndpoint.
+ * It aggregates the status across all Plans and allows
+ * for filtered responses. The PlansTracker also returns
+ * the currently active plans and the current state the
+ * the scheduler is in.
+ */
+public class PlansTracker implements DebugEndpoint {
+
+  private final StateStore stateStore;
+
+  private final PlanCoordinator planCoordinator;
+
+  public PlansTracker(PlanCoordinator planCoordinator, StateStore stateStore) {
+    this.planCoordinator = planCoordinator;
+    this.stateStore = stateStore;
+  }
+
+  private JSONObject generateServiceStatus(String filterPlan,
+                                           String filterPhase,
+                                           String filterStep,
+                                           boolean allowFiltering,
+                                           boolean skipDetails)
+  {
```

Comment:
inconsistent with the `{` opening... all previous to this are end of line... this is beginning of line.   can we have all end of line?

Reply:
IIRC it was on the same line before, but checkstyle didn't like that. Same goes for many other ones below.